File: app/src/test/java/com/kuanhsien/app/sample/android_mvvm_demo/BaseTest.java
package com.kuanhsien.app.sample.android_mvvm_demo;

import androidx.arch.core.executor.testing.InstantTaskExecutorRule;
import org.junit.Before;
import org.junit.Rule;
import org.mockito.MockitoAnnotations;


public class BaseTest {

    // Execute everything in foreground thread (synchronized)
    @Rule
    public InstantTaskExecutorRule instantTaskExecutorRule = new InstantTaskExecutorRule();

    @Before
    public void setUp() {
        MockitoAnnotations.initMocks(this);
    }
}

File: app/src/test/java/com/kuanhsien/app/sample/android_mvvm_demo/BaseTest.kt
package com.kuanhsien.app.sample.android_mvvm_demo

import io.mockk.MockKAnnotations
import io.mockk.unmockkAll
import org.junit.After
import org.junit.Before

class BaseTest {

    @Before
    open fun init() {
        // For MockK annotation
        MockKAnnotations.init(this, relaxed = true)
        // or MockKAnnotations.init(this, relaxUnitFun = true)
    }

    @After
    fun after() {
        unmockkAll()
        // or unmockkObject(MockObj)
    }
}